File: module_stu/src/main/java/com/sunny/module/stu/fragment/StuFragmentSelector.java
package com.sunny.module.stu.fragment;

import com.sunny.module.stu.base.StuBaseFragment;
import com.sunny.module.stu.handler.StuHandlerFragment;

import java.util.HashMap;
import java.util.Map;

public class StuFragmentSelector {

    private static final Map<String, StuBaseFragment> mFragmentMap = new HashMap<>();

    public static StuBaseFragment getFragment(String type) {

        StuBaseFragment result = mFragmentMap.get(type);
        if (result != null) {
            return result;
        }

        switch (type) {
            case "stu_handler":
                result = new StuHandlerFragment();
                break;

            default:
                result = new NormalFragment();
                break;
        }

        mFragmentMap.put(type, result);
        return result;
    }
}

File: module_stu/src/main/java/com/sunny/module/stu/fragment/StuFragmentSelector.kt
package com.sunny.module.stu.fragment

import com.sunny.module.stu.base.StuBaseFragment
import com.sunny.module.stu.git.StuGitFragment
import com.sunny.module.stu.handler.StuHandlerFragment
import java.util.*

object StuFragmentSelector {
    private val mFragmentMap: MutableMap<String, StuBaseFragment> = HashMap()

    @JvmStatic
    fun getFragment(type: String): StuBaseFragment {
        var result = mFragmentMap[type]
        if (result != null) {
            return result
        }

        //
        result = when (type) {
            "stu_handler" -> StuHandlerFragment()
            "stu_git" -> StuGitFragment()
            else -> NormalFragment()
        }
        mFragmentMap[type] = result
        return result
    }
}